lib/axon/loop.ex

```diff
@@ -1304,6 +1304,64 @@ defmodule Axon.Loop do
     )
   end
 
+  @doc """
+  Adds a handler function which updates a `VegaLite`.
+
+  By default, this will run after every iteration.
+
+  You must specify an empty line plot to push to. By default, this will monitor your loop's loss, but you may specify a different metric. The `:y` axis will be the iteration count, labeled `"step"`. The metric must match the name given to the `:x` axis in your `VegaLite` plot:
+
+      plot =
+        Vl.new()
+        |> Vl.mark(:line)
+        |> Vl.encode_field(:x, "step", type: :quantitative)
+        |> Vl.encode_field(:y, "loss", type: :quantitative)
+        |> Kino.VegaLite.new()
+        |> Kino.render()
+
+      model
+      |> Axon.Loop.trainer(loss, optim)
+      |> Axon.Loop.plot(plot)
+  """
+  def plot(loop, plot, metric \\ "loss", event \\ :iteration_completed, filter \\ :always) do
```

Comment:
```suggestion
  def plot(loop, plot, opts \\ []) do
```

and set those defaults with `Keyword.validate!`?

Reply:
+1, but I would make `metric` a required argument!

Reply:
So like this?

```suggestion
  def plot(loop, plot, metric, opts \\ []) do
```

Reply:
Yep!